src/main/java/org/broadinstitute/hellbender/utils/bigquery/StorageAPIAvroReader.java

```diff
@@ -0,0 +1,161 @@
+package org.broadinstitute.hellbender.utils.bigquery;
+
+import com.google.cloud.bigquery.storage.v1beta1.*;
+import com.google.common.base.Preconditions;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.BinaryDecoder;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public class StorageAPIAvroReader implements GATKAvroReader {
```

Comment:
If this class is used in subsequent branches, we'll need to add a test for it. A basic unit test for `BigQueryUtils` that exercises the (currently commented-out) `executeQueryWithStorageAPI()` method would suffice.